tools/worker-runner/errorreport/errorreport_test.go

```diff
@@ -0,0 +1,102 @@
+package errorreport
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/taskcluster/taskcluster/v30/clients/client-go/tcworkermanager"
+	"github.com/taskcluster/taskcluster/v30/internal/workerproto"
+	ptesting "github.com/taskcluster/taskcluster/v30/internal/workerproto/testing"
+	"github.com/taskcluster/taskcluster/v30/tools/worker-runner/run"
+	"github.com/taskcluster/taskcluster/v30/tools/worker-runner/tc"
+)
+
+func init() {
+	workerManagerClientFactory = tc.FakeWorkerManagerClientFactory
+}
+
+func TestHandleMessage(t *testing.T) {
+	description := "this is a serious error"
+	extra := map[string]interface{}{
+		"foo": "bar",
+	}
+	kind := "severe"
+	title := "test error"
+
+	t.Run("with capability", func(t *testing.T) {
+		wkr := ptesting.NewFakeWorkerWithCapabilities("error-report")
+		defer wkr.Close()
+
+		state := run.State{}
+		wkr.RunnerProtocol.Register("error-report", func(msg workerproto.Message) {
+			HandleMessage(msg, workerManagerClientFactory, &state)
+		})
+		wkr.RunnerProtocol.AddCapability("error-report")
+		wkr.RunnerProtocol.Start(false)
+		wkr.RunnerProtocol.WaitUntilInitialized()
+		wkr.WorkerProtocol.Start(true)
+		wkr.WorkerProtocol.Send(workerproto.Message{
+			Type: "error-report",
+			Properties: map[string]interface{}{
+				"description": description,
+				"extra":       extra,
+				"kind":        kind,
+				"title":       title,
+			},
+		})
+
+		var reports []*tcworkermanager.WorkerErrorReport
+		var err error
+		func() {
```

Comment:
Why is this in a goroutine?  Wouldn't it fail while a subsequent test is running?

BTW I added some flushing support in 40caa12290 which will hopefully make this easier.

Reply:
It's an anonymous function that gets called, it isn't a goroutine (no `go`), so it runs inline

Reply:
Oh, I did miss that!  But, I am still unclear on the reason to do so..

Reply:
I did it so that I could neatly return or panic from that function rather than break the loop

Reply:
Ah, I see.. neat!